crates/chia-sdk-driver/src/layers/datalayer/oracle_layer.rs

```diff
@@ -0,0 +1,94 @@
+use chia_protocol::{Bytes, Bytes32};
+use chia_sdk_types::{Condition, CreateCoin, CreatePuzzleAnnouncement};
+use clvm_traits::{clvm_quote, match_quote, FromClvm, ToClvm};
+use clvmr::{Allocator, NodePtr};
+
+use crate::{DriverError, Layer, Puzzle, Spend, SpendContext};
+
+/// The Oracle [`Layer`] enables anyone to spend a coin provided they pay an XCH fee to an address.
+/// It's typically used with [`DelegationLayer`](crate::DelegationLayer).
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct OracleLayer {
+    /// The puzzle hash corresponding to the address the fee should be paid to.
+    pub oracle_puzzle_hash: Bytes32,
+    /// The amount of XCH that should be paid to the oracle.
+    pub oracle_fee: u64,
+}
+
+impl OracleLayer {
+    #[allow(clippy::missing_panics_doc)]
+    pub fn new(oracle_puzzle_hash: Bytes32, oracle_fee: u64) -> Self {
+        assert!(oracle_fee % 2 == 0, "oracle fee must be even");
+
+        Self {
+            oracle_puzzle_hash,
+            oracle_fee,
+        }
+    }
+}
+
+impl Layer for OracleLayer {
+    type Solution = ();
+
+    fn parse_puzzle(allocator: &Allocator, puzzle: Puzzle) -> Result<Option<Self>, DriverError> {
+        let Some(puzzle) = puzzle.as_raw() else {
+            return Ok(None);
+        };
+
+        let (_q, conditions) =
+            <match_quote!(Vec<Condition<NodePtr>>)>::from_clvm(allocator, puzzle.ptr)?;
+        if conditions.len() != 2 {
```

Comment:
Is this and the ordering not too restrictive? You could iterate through the conditions like this instead for example:
```rs
let create_coin = conditions.into_iter().filter_map(Condition::into_create_coin).find(|create_coin| { 
/* maybe odd or something */ });
```

Reply:
I would be on the stricter side given that the CHIP specifies the exact puzzle & `OracleLayer` builds it this way - if we stray away from this, hints for oracles might become meaningless as observers/parsers aren't sure how to reconstruct the puzzle